src/Compress/FileExtensionArchiveAdapterResolver.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Laminas\Filter\Compress;
+
+use Laminas\Filter\Exception\RuntimeException;
+use Laminas\Filter\File\FileInformation;
+
+use function basename;
+use function pathinfo;
+use function sprintf;
+use function str_ends_with;
+use function strtolower;
+
+use const PATHINFO_EXTENSION;
+
+final class FileExtensionArchiveAdapterResolver implements ArchiveAdapterResolverInterface
```

Comment:
Simply matches `whatever.zip` to use the Zip archive adapter based on filename extension. By default it is composed last in the chain, preferring mime-magic based mime detection.